src/gui/accountsettings.cpp

```diff
@@ -888,13 +893,65 @@ AccountSettings::~AccountSettings()
     delete _ui;
 }
 
+void AccountSettings::slotHideSelectiveSyncWidget(){
+    _ui->selectiveSyncApply->setEnabled(false);
+    _ui->selectiveSyncStatus->setVisible(false);
+    _ui->selectiveSyncButtons->setVisible(false);
+    _ui->selectiveSyncLabel->hide();
+}
+
+void AccountSettings::slotSelectiveSyncChanged(const QModelIndex &topLeft,
+                                      const QModelIndex &bottomRight,
+                                      const QVector<int> &roles){
+    Q_UNUSED(bottomRight);
+    if (roles.contains(Qt::CheckStateRole)) {
+        const auto info = _model->infoForIndex(topLeft);
+        if (info) {
+            const bool showWarning = _model->isDirty() && _accountState->isConnected() && info->_checked == Qt::Unchecked;
+
+            // FIXME: the model is not precise enough to handle extra cases
+            // e.g. the user clicked on the same checkbox 2x without applying the change in between.
+            // We don't know which checkbox changed to be able to toggle the selectiveSyncLabel display.
+            if (showWarning) {
+                _ui->selectiveSyncLabel->show();
+            }
+
+            const bool shouldBeVisible = _model->isDirty();
+            const bool wasVisible = _ui->selectiveSyncStatus->isHidden();
+            if (shouldBeVisible) {
+                _ui->selectiveSyncStatus->setVisible(true);
+            }
+
+            _ui->selectiveSyncApply->setEnabled(true);
+            _ui->selectiveSyncButtons->setVisible(true);
+
+            if (shouldBeVisible != wasVisible) {
+                const auto hint = _ui->selectiveSyncStatus->sizeHint();
+
+                if (shouldBeVisible) {
+                    _ui->selectiveSyncStatus->setMaximumHeight(0);
+                }
+
+                const auto anim = new QPropertyAnimation(_ui->selectiveSyncStatus, "maximumHeight", _ui->selectiveSyncStatus);
+                anim->setEndValue(_model->isDirty() ? hint.height() : 0);
+                anim->start(QAbstractAnimation::DeleteWhenStopped);
+                connect(anim, &QPropertyAnimation::finished, [this, shouldBeVisible]() {
+                    _ui->selectiveSyncStatus->setMaximumHeight(QWIDGETSIZE_MAX);
+                    if (!shouldBeVisible) {
+                        _ui->selectiveSyncStatus->hide();
+                    }
+                });
+            }
+        }
+    }
+}
+
 void AccountSettings::refreshSelectiveSyncStatus()
 {
-    bool shouldBeVisible = _model->isDirty() && _accountState->isConnected();
-
-    QString msg;
+    QString bigFolderMsg;
```

Comment:
I think I wouldn't have renamed that variable just for the smaller patch which would ensue. Especially since it's tangentially relevant to the rest of the commit.